app/presenters/self_service/status_presenter.rb

```diff
@@ -0,0 +1,82 @@
+module SelfService
+  class StatusPresenter
+    include Rails.application.routes.url_helpers
+    include ActionView::Helpers::UrlHelper
+
+    def initialize(firm)
+      @firm = firm
+    end
+
+    def overall_status
+      @firm.publishable? ? 'published' : 'unpublished'
+    end
+
+    def overall_status_icon
+      icon_toggle @firm.publishable?
+    end
+
+    def firm_details_icon
+      icon_toggle @firm.registered?
+    end
+
+    def advisers_icon
+      icon_toggle(remote? || @firm.advisers.any?)
+    end
+
+    def offices_icon
+      icon_toggle @firm.offices.any?
+    end
+
+    def firm_details_link(opts = {})
+      path = if @firm.trading_name?
+               edit_self_service_trading_name_path(@firm)
+             else
+               edit_self_service_firm_path(@firm)
+             end
+
+      link_to I18n.t('self_service.firms_index.status.edit'), path, opts
+    end
+
+    def advisers_link(opts = {})
+      if @firm.advisers.present?
+        path = self_service_firm_advisers_path(@firm)
+        text = I18n.t('self_service.firms_index.status.edit')
+      else
+        path = new_self_service_firm_adviser_path(@firm)
+        text = I18n.t('self_service.firms_index.status.add')
+      end
+
+      link_to text, path, opts
+    end
+
+    def offices_link(opts = {})
+      if @firm.offices.present?
+        path = self_service_firm_offices_path(@firm)
+        text = I18n.t('self_service.firms_index.status.edit')
+      else
+        path = new_self_service_firm_office_path(@firm)
+        text = I18n.t('self_service.firms_index.status.add')
+      end
+
+      link_to text, path, opts
+    end
+
+    def advisers_count
+      @firm.advisers.count
+    end
+
+    def offices_count
+      @firm.offices.count
+    end
+
+    private
+
+    def icon_toggle(condition_met)
+      condition_met ? 'tick' : 'exclamation'
+    end
+
+    def remote?
+      @firm.primary_advice_method == :remote
+    end
+  end
+end
```

Comment:
This is good :+1: Just for the sake of passing knowledge you can inherit from SimpleDelegator and act exactly like a Presenter and proxying the object on initialize.